### ISSUE TITLE
CodeKit launches constantly even with no projects defined

### ISSUE THREAD

**Issue by dcsjapan** [bdkjones/CodeKitNova#4]
I have the extension set to:

- Auto-Add Projects: Never
- Auto-Switch CodeKit Projects: Yes

I have no projects in CodeKit, hence no Nova projects containing any CodeKit config info. Logically there should be no reason for the extension to launch CodeKit no matter how many times I switch between projects. Yet it does, constantly.

It seems to be happening when I open projects in Nova. It's as if the extension is launching CodeKit before it decides whether it needs to take action.

Update: I uninstalled the extension and CodeKit stopped launching when I opened / switched between Nova projects.

**Comment by bdkjones** [comment]
This is a limitation of the API, currently. Nothing gets added to CodeKit if there's no config file in the Nova project you open, but the app itself starts running if it's not already active. I'd like to eliminate that as well.

**Comment by dcsjapan** [comment]
Thanks for responding. I suppose this doesn't matter much to programmers who've got CodeKit running all the time, but I got CodeKit mainly to compile SASS on a few projects and don't need it for all my projects.

Here's hoping the good folks at Panic can make those changes to the API. More (and better) extensions can only be a good thing for Nova's future.

**Comment by bdkjones** [comment]
Yep. If I recall correctly, Nova fires up CodeKit anytime I check to see what you have the plugin configured to do. This line:

var autoAddProject = nova.config.get('com.CodeKitApp.Nova.autoAddProjects’);

appears to fire up the app, even though all I’m doing is loading the configuration to see if you have the plugin set to auto-add projects that have a config file, ALL projects, or no projects. It’s as if Nova is attempting to verify that CodeKit is available on the system by launching it.

**Comment by dcsjapan** [comment]
That would certainly account for the behavior I noticed. It doesn't really make sense, does it? The extension should be able to check its own settings regardless of whether a third party app is even installed, much less whether it has been launched.
